lib/l10n/l10n.dart

```diff
@@ -98,10 +98,22 @@ class Language {
 
   /// Returns a [Language] identified by its [tag] from the [L10n.languages], if
   /// any.
-  static Language? from(String? tag) {
+  static Language? fromTag(String? tag) {
     return L10n.languages.firstWhereOrNull((e) => e.toString() == tag);
   }
 
+  /// Returns a [Language] from the [L10n.languages] matching the provided
+  /// [locale], if any.
+  static Language? fromLocale(Locale locale) {
+    return L10n.languages.firstWhereOrNull((e) => e.locale == locale) ??
+        // If exact match wasn't found, then find any matching the language or
+        // country code.
+        L10n.languages.firstWhereOrNull(
+            (e) => e.locale.languageCode == locale.languageCode) ??
+        L10n.languages.firstWhereOrNull(
+            (e) => e.locale.countryCode == locale.countryCode);
+  }
```

Comment:
@tyranron, parsing is done by the [`WidgetsBinding.instance.platformDispatcher.locales`](https://api.flutter.dev/flutter/dart-ui/SingletonFlutterWindow/locales.html) getter, and the problem lies only within our comparison of `L10n.languages` with the parsed `Locale` returned by the getter. Thus I suggest the highlighted solution, which is accounting the language code as well as the country code. We hold both in the `L10n.languages`:

```
static List<Language> languages = const [
  Language('English', Locale('en', 'US')),
  Language('Русский', Locale('ru', 'RU')),
];
```

Reply:
@SleepySquash why "reinvent the wheel"? Why not just [`basicLocaleListResolution`](https://api.flutter.dev/flutter/widgets/basicLocaleListResolution.html)?

Reply:
@tyranron, thank you for the link! Nice catch, the algorithm you provided seems to be even more compicated than a simple `for` presented in this PR. I'll reimplement the PR to invoke this instead.